test/authenticate.spec.js

```diff
@@ -11,46 +11,50 @@ chai.should();
 chai.use(sinonChai);
 
 describe('authenticateToken', () => {
-	const publickeyStub = sinon.stub();
+	const publickeyPollerStub = sinon.stub();
 	const validateStub = sinon.stub();
-	const authenticate = proxyquire('../authenticate', {
-		'./publickey': () => publickeyStub,
+	const authenticateFactory = proxyquire('../authenticate', {
 		'./validate': () => validateStub,
-	}).authenticateToken;
+	});
+	const getAuthenticateToken = () => authenticateFactory(publickeyPollerStub).authenticateToken;
 
 	beforeEach(() => {
-		publickeyStub.reset()
-		publickeyStub.returns(() => true);
+		publickeyPollerStub.reset();
```

Comment:
why is publickey used instead of publicKey everywhere in this file?

Reply:
Changed everywhere to `publickey` to match the export.